File: leetcode/1.两数之和.java
/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[]result=new int[2];
        HashMap<Integer, Integer>map=new HashMap<Integer, Integer>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i],i);
        }
        for(int i=0;i<nums.length;i++){
            int temp=target-nums[i];
            if(map.containsKey(temp)&&map.get(temp)!=i){
                result[0]=i;
                result[1]=map.get(temp);
                break;
            }
        }
        return result;
   }
}
// @lc code=end

File: leetcode/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mapping;
        vector<int> result;
        for(int i = 0; i < nums.size();i++){
            mapping[nums[i]]=i;
        }
        for(int i=0;i<nums.size();i++){
            int temp=target-nums[i];
            if(mapping.find(temp)!=mapping.end()&&mapping[temp]>i){
                result.push_back(i);
                result.push_back(mapping[temp]);
                return result;
            }
        }
        return result;
    }
};
// @lc code=end

